### PR TITLE
Add base config for auth0, create new project base

### PR DESCRIPTION
**Pull request by KamilSupera**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,34 @@
...
+The project needs to be configured with your Auth0 domain and client ID in order for the authentication flow to work.
+
+To do this, first copy `auth_config.json.example` into a new file in the same folder called `auth_config.json`, and replace the values within with your own Auth0 application credentials:
+
+```json
+{
+  "domain": "<YOUR AUTH0 DOMAIN>",
+  "clientId": "<YOUR AUTH0 CLIENT ID>"
+}
+```
```
Comment (kaniak274) [inline README.md]: I think we don't need that. Just hardcode credentials in javascript file.
Rationale: These are not the secrets so we don't have to care about security

File: src/auth/index.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
+import Vue from "vue";
```
Comment (kaniak274) [inline src/auth/index.js:1]: Are you sure you want to use two spaces? I don't mind but remember to be consistent in other files.

File: src/router/index.js [hunk 3]
```diff
@@ -1,9 +1,21 @@
...
+import VueRouter from 'vue-router'
+import Home from '../views/Home.vue'
 
-Vue.use(Router)
+Vue.use(VueRouter)
 
-export default new Router({
-    routes: [
-    ]
+  const routes = [
```
Comment (kaniak274) [inline src/router/index.js]: To much spaces

File: src/auth/index.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+}) => {
+  if (instance) return instance;
+
+  // The 'instance' is simply a Vue object
+  instance = new Vue({
+    data() {
+      return {
+        loading: true,
+        isAuthenticated: false,
+        user: {},
```
Comment (kaniak274) [inline src/auth/index.js]: Is this connected to vuex store? If not how are you going to get this information in other views. I guess information like username or email may be useful to show to user after login for example.

File: src/views/Home.vue [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        <!-- Check that the SDK client is not currently loading before accessing is methods -->
+        <div v-if="!$auth.loading">
+
+            <!-- show login when not authenticated -->
+            <button v-if="!$auth.isAuthenticated" @click="login">Log in</button>
+            <div v-if="$auth.isAuthenticated">
+                <h1 v-if="$auth.isAuthenticated">{{ $auth.user.name }}</h1>
+
+                <!-- shwo token after generate -->
+                <div> {{ token }} </div>
```
Comment (kaniak274) [inline src/views/Home.vue:12]: Remember to remove it later.

File: src/App.vue [hunk 6]
```diff
@@ -1,17 +1,8 @@
 <template>
   <div id="app">
-    <img src="./assets/images/a.jpg">
-    <h1>{{ msg }}</h1>
+    <div id="nav">
+      <router-link to="/">Home</router-link>
```
Comment (kaniak274) [inline src/App.vue]: Please use named routes.

```
<router-link to="{ name: 'home' }" ...
```

**Review by kaniak274 — CHANGES_REQUESTED** [review]

File: src/auth/index.js (hunk 2 above)
Comment (KamilSupera) [reply src/auth/index.js:1]: These, functions etc... is generated from auth0, so I think is okay?

File: src/auth/index.js (hunk 4 above)
Comment (KamilSupera) [reply src/auth/index.js]: For now, I add only pure base vue. Also all user data is stored under global $auth.user

Comment (kaniak274) [reply src/auth/index.js]: Oh so it's fine then

File: src/auth/index.js (hunk 2 above)
Comment (kaniak274) [reply src/auth/index.js:1]: As I said I don't mind (even tho I think four spaces are better) but use two spaces in other files as well

**Comment by kaniak274** [comment]
I've merge the bump pr and conflicts arise please fix them.

**Review by kaniak274 — APPROVED** [review]
I will check later if this work and then merge. For know I'll give you an approve

**Pull request merged**